db/blip_sync_context.go

```diff
@@ -354,12 +368,18 @@ func (bsc *BlipSyncContext) handleChangesResponse(sender *blip.Sender, response
 				}
 			}
 
-			var err error
-			if deltaSrcRevID != "" {
-				err = bsc.sendRevAsDelta(sender, docID, revID, deltaSrcRevID, seq, knownRevs, maxHistory, handleChangesResponseDbCollection, collectionIdx)
-			} else {
-				err = bsc.sendRevision(sender, docID, revID, seq, knownRevs, maxHistory, handleChangesResponseDbCollection, collectionIdx)
-			}
+			err := func() error {
```

Comment:
Given that the below callbacks are only called when ISGR is in place, does it make sense to drop this function here and put the defer until the end of `handleChangesResponse` including the possible callbacks?

The rest of the time spent in `handleChangesResponse` is not in sending a message back so it won't overwhelm the network here so I see a good reason not to include the callbacks in the throttling here, but I worry about a potential refactor meaning that we are throttling too little.